externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/spark/kyuubi/SQLOperationListener.scala

```diff
@@ -140,6 +149,14 @@ class SQLOperationListener(
     val stageInfo = stageCompleted.stageInfo
     val stageId = stageInfo.stageId
     val stageAttempt = SparkStageAttempt(stageInfo.stageId, stageInfo.attemptNumber())
+    val taskMetrics = stageInfo.taskMetrics
```

Comment:
Similar to https://github.com/apache/kyuubi/pull/6113#discussion_r1508508432

Reply:
Each `SQLOperationListener` is created for each `ExecuteStatement`.
So the events received by the `SQLOperationListener` belong to that operation.
Therefore we don't need to determine whether a stage belongs to a operation.

https://github.com/apache/kyuubi/blob/master/externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/ExecuteStatement.scala#L98

Reply:
> So the events received by the SQLOperationListener belong to that operation.

This doesn't seem to ensure.

We just need to move this change into the following  closure.

https://github.com/apache/kyuubi/blob/9f53a09d5830801c4f3215a2e210812d1de8ef5d/externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/spark/kyuubi/SQLOperationListener.scala#L144

Reply:
`SparkListenerBus` dispatches all spark events. We made a `sameGroupId` judgment when processing the startup events to ensure that the events processed belong to current operation.

Reply:
But using the sameGroupId will miss broadcast operators.  
https://github.com/apache/kyuubi/blob/9f53a09d5830801c4f3215a2e210812d1de8ef5d/externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/spark/kyuubi/SQLOperationListener.scala#L66C1-L72C4

Reply:
> But using the sameGroupId will miss broadcast operators. 

This is not a reason not to determine whether stage belongs to current operation.

Reply:
moved this change into the closure